src/include/message.h

```diff
@@ -55,6 +55,26 @@ struct message
    void* data;        /**< The message data */
 } __attribute__ ((aligned (64)));
 
+/** @struct
+ * Defines a tuple
+ */
+struct tuple
+{
+   char** data;                   /**< The data */
+   struct tuple* next;            /**< The next tuple */
+} __attribute__ ((aligned (64)));
+
+/** @struct
+ * Defines a query
+ */
+struct query
+{
+   char names[MAX_NUMBER_OF_COLUMNS][MISC_LENGTH]; /**< The column names */
+   int number_of_columns;                          /**< The number of columns */
+
+   struct tuple* tuples;
+} __attribute__ ((aligned (64)));
```

Comment:
I'm not sure what this means. Does the struct contains all the tuples the query touches?